### DISCUSSION TITLE
Increase Dead set limit above default 10K or 6 months

### DISCUSSION THREAD

**Discussion by iamitpatil1993** [sidekiq/sidekiq#5011] [Q&A] [answered]
Hello Team,

We use Sidekiq in one of our business-critical service. Recently, we faced one production incident which last for few hours.
So, our dead got full, with default limit of 10K.
So, we wanted to check how we can increase this limit, we have enough space in Redis which is being used for Sidekiq. And what happens the jobs tried to be enqueued when deadset is full? Do we lose these jobs.

We want to process all these jobs, so to be safer for the next time, we want to increase the deadset.
We checked the docs and FAQs as well but, could not find any config which we can use to extend the deadset max limit.

Please help us with this. Will highly appropriate any help here.
Thanks in advance

**Comment by mperham** [comment, marked as answer]
The Retry set is unlimited, maybe you should not disable retries?

```ruby
Sidekiq.configure_server do |config|
  config.options[:dead_max_jobs] = 100_000
end
```

If a popular job is not important to keep around, set `sidekiq_options dead: false` so any failed instances do not quickly fill up your Dead set.

**Reply by iamitpatil1993** [reply]
Thanks, @mperham for the quick response

> maybe you should not disable retries?
Yup, we have retried enabled already.

Tried this config and it works well. Thanks again 🙏 :)

**Reply by deivinsontejeda** [reply]
> The Retry set is unlimited, maybe you should not disable retries?
> 
> ```ruby
> Sidekiq.configure_server do |config|
>   config.options[:dead_max_jobs] = 100_000
> end
> ```
> 
> If a popular job is not important to keep around, set `sidekiq_options dead: false` so any failed instances do not quickly fill up your Dead set.

Does this work for Sidekiq 4.2?

**Reply by jeffbax** [reply]
I just attempted this with v7, and got `undefined method 'options' for #<Sidekiq::Config:0x00007f224b1e3cb8>` so I'm not sure if this was working code or potential code.

**Reply by mperham** [reply]
Sidekiq 7:

```ruby
Sidekiq.default_configuration[:dead_max_jobs] = 100_000
```

**Reply by jeffbax** [reply]
just finally tried that myself :) thank you
